pkgs/development/python-modules/pytest-xdist/default.nix

```diff
@@ -25,9 +26,11 @@ buildPythonPackage rec {
   buildInputs = [
     pytest
   ];
-  checkInputs = [ pytestCheckHook filelock ];
+  checkInputs = [ pytestCheckHook filelock pexpect ];
   propagatedBuildInputs = [ execnet pytest-forked psutil ];
 
+  pytestFlagsArray = [ "-n $NIX_BUILD_CORES" "--forked" ]; # pytest can already use xdist at this point
```

Comment:
Doesn't this duplicate what happens in the setup hook?

Reply:
i don't think the setup hook is used at this point yet?

Reply:
just tested and yep its not used at this point



if i apply the following the build fails with
`python3.9-pytest-xdist> substitute(): ERROR: file '/nix/store/q4qj7q82vrxfwnmymyhdwxid0sr0rjnn-hook' does not exist`

```patch
diff --git a/pkgs/development/python-modules/pytest-xdist/default.nix b/pkgs/development/python-modules/pytest-xdist/default.nix
index 75fd066c76d..ae23bf3f31b 100644
--- a/pkgs/development/python-modules/pytest-xdist/default.nix
+++ b/pkgs/development/python-modules/pytest-xdist/default.nix
@@ -1,4 +1,5 @@
 { lib
+, makeSetupHook
 , buildPythonPackage
 , fetchPypi
 , pythonOlder
@@ -12,6 +13,10 @@
 , pexpect
 }:
 
+let
+  setupHook' = makeSetupHook {} ./setup-hook.sh;
+in
+
 buildPythonPackage rec {
   pname = "pytest-xdist";
   version = "2.5.0";
@@ -26,11 +31,9 @@ buildPythonPackage rec {
   buildInputs = [
     pytest
   ];
-  checkInputs = [ pytestCheckHook filelock pexpect ];
+  checkInputs = [ pytestCheckHook filelock pexpect setupHook' ];
   propagatedBuildInputs = [ execnet pytest-forked psutil ];
 
-  pytestFlagsArray = [ "-n $NIX_BUILD_CORES" "--forked" ]; # pytest can already use xdist at this point
-
   # access file system
   disabledTests = [
     "test_distribution_rsyncdirs_example"
@@ -43,7 +46,7 @@ buildPythonPackage rec {
     "test_internal_errors_propagate_to_controller"
   ];
 
-  setupHook = ./setup-hook.sh;
+  setupHook = setupHook';
 
   meta = with lib; {
     description = "Pytest xdist plugin for distributed testing and loop-on-failing modes";
```